internal/elasticsearch/cluster/component_template.go

```diff
@@ -0,0 +1,290 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/elasticsearch/index"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceComponentTemplate() *schema.Resource {
+	// NOTE: component_template and index_template uses the same schema
+	componentTemplateSchema := map[string]*schema.Schema{
```

Comment:
Please, also add the ID as `Computed` value:

```suggestion
    componentTemplateSchema := map[string]*schema.Schema{
			"id": {
			Description: "Internal identifier of the resource",
			Type:        schema.TypeString,
			Computed:    true,
		},
```